### PR TITLE
profile api exception handling

### PR DESCRIPTION
**Pull request by zasyadev**

### CODE REVIEW

File: component/Profile/Profile.jsx [hunk 1]
```diff
@@ -40,14 +40,11 @@ function Profile({ user, previewMode = false }) {
     await httpService
       .get(`/api/profile/${userId}`)
       .then(({ data: response }) => {
-        if (response.status === 200) {
-          setUserDetails(response.data);
-        }
+        setUserDetails(response.data);
         setLoading(false);
```
Comment (kshitijzasya) [inline component/Profile/Profile.jsx]: remove this if you are using finally

File: component/Profile/Profile.jsx [hunk 2]
```diff
@@ -40,14 +40,11 @@ function Profile({ user, previewMode = false }) {
...
       })
-      .catch((err) => {
-        setLoading(false);
-      });
+      .catch(() => setUserDetails({}))
```
Comment (kshitijzasya) [inline component/Profile/Profile.jsx]: setUserDetails should be null. don't set empty object

File: pages/api/profile/[user_id].js [hunk 3]
```diff
@@ -1,20 +1,17 @@
...
     gte: moment().startOf("month").format(),
   };
 
-  if (!user_id) {
-    return res.status(401).json({ status: 401, message: "No User found" });
-  }
+  if (!user_id) throw BadRequestException("No User found");
 
   if (req.method === "GET") {
     const whereCondition = {
```
Comment (kshitijzasya) [inline pages/api/profile/[user_id].js]: use proper naming for this variable

Comment (zasyadev) [reply pages/api/profile/[user_id].js]: what are you asking to change "BadRequestException" 
as we have created a common exception for all the Bad request and sending just a string to it

Comment (kshitijzasya) [reply pages/api/profile/[user_id].js]: const whereCondition = { this varible name is not suitable

**Pull request merged**